packages/common/src/account/kms/createKmsAccount.ts

```diff
@@ -25,38 +25,46 @@ export async function createKmsAccount({
   const account = toAccount({
     address,
     async signMessage({ message }) {
-      const hash = hashMessage(message);
       const signature = await signWithKms({
         client,
         keyId,
-        hash,
+        hash: hashMessage(message),
         address,
       });
 
-      return signature;
+      return signatureToHex(signature);
     },
-    async signTransaction(transaction) {
-      const unsignedTx = serializeTransaction(transaction);
-      const hash = keccak256(unsignedTx);
+    async signTransaction(transaction, { serializer = serializeTransaction } = {}) {
+      // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+      const signableTransaction = (() => {
+        // For EIP-4844 Transactions, we want to sign the transaction payload body (tx_payload_body) without the sidecars (ie. without the network wrapper).
+        // See: https://github.com/ethereum/EIPs/blob/e00f4daa66bd56e2dbd5f1d36d09fd613811a48b/EIPS/eip-4844.md#networking
+        if (transaction.type === "eip4844")
+          return {
+            ...transaction,
+            sidecars: false,
+          };
+        return transaction;
+      })();
+
       const signature = await signWithKms({
         client,
         keyId,
-        hash,
+        hash: keccak256(serializer(signableTransaction)),
         address,
       });
 
-      return signature;
+      return serializer(transaction, signature);
```

Comment:
just to confirm, we _don't_ want to use `signableTransaction` here?

Reply:
Nope, again this is [from viem](https://github.com/wevm/viem/blob/main/src/accounts/utils/signTransaction.ts#L68-L71). I think it's because the data you sign should (or shouldn't idk) include some data that is in the full transaction data.